### PR TITLE
Add data_level and data_sublevel keywords for ObsSurface

### PR DESCRIPTION
**Pull request by rt17603**
* **Summary of changes** (Bug fix, feature, docs update, ...)

When standardising observation surface data, this adds `data_level` and `data_sublevel` keys as options to define the data.

Notes:
 - Adds these keys as required keys for data_type="surface"
     - Uses `"not_set"` as the default when they are not specified
 - data_level keyword is checked to ensure this is only "0", "1", "2", or "3" (also allows decimals)
 - This is a basic implementation of this which is explicit about which keys are (a) not being passed to the parse function and then (b) being added to the metadata afterwards. The next step would be to improve this to make it more dynamic (next step for this is #1069)

* **Please check if the PR fulfills these requirements**

- [x] Addresses Issue #551
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature

Not done:
- Documentation and tutorials updated/added
- Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

### CODE REVIEW

File: openghg/util/_data_level.py
Comment (brendan-m-murphy) [inline openghg/util/_data_level.py:48]: Maybe it's worth using `is_number` from `util`, like 

``` 
if not is_number(data_level):
   raise ValueError(...)
number = float(data_level)
```

File: openghg/util/_strings.py
Comment (brendan-m-murphy) [inline openghg/util/_strings.py:175]: This is a nice additional. Eventually it should go with the rest of the metadata formatting stuff, but I'll just make a note of that.

File: openghg/store/base/_base.py
Comment (brendan-m-murphy) [inline openghg/store/base/_base.py:163]: Maybe call this something else since we're updated the metadata dictionary, not just the keys. Maybe just `update_metadata`?

**Review by brendan-m-murphy — CHANGES_REQUESTED** [review]
Looks good! Just a few small suggestions.

File: openghg/standardise/_standardise.py
Comment (rt17603) [reply openghg/standardise/_standardise.py:95]: I've toyed with with the idea of using Enumerate / Literal for these types of inputs but generally came to the conclusion that I didn't want to make the interface for complex for the user - fine if we could allow the user to pass a string still though. I'm going to resolve this for now, if that's ok, but something to come back to with the metadata formatting.

File: openghg/store/base/_base.py
Comment (rt17603) [reply openghg/store/base/_base.py:163]: You're right we should be consistent - I included `Dict` for this as that was what the `data` input to `assign_data` was so I'm actually going to update it to:
```suggestion
    def add_metadata_keys(self, data: Dict, additional_metadata: Dict) -> Dict:
```

File: openghg/util/_data_level.py
Comment (rt17603) [reply openghg/util/_data_level.py:45]: It was actually for a silly corner case which does come up where even though our type hinting asks for a string it doesn't enforce it so sometimes other types do get passed - this was to account for the integer case.
```suggestion
        data_level = str(data_level)  # in case a int/float is passed
```

File: openghg/util/_strings.py
Comment (rt17603) [reply openghg/util/_strings.py:175]: Yeah agreed - I was struggling to decide where to put it so kept with strings for now but will be better to include somewhere central when we pull together formatting.

File: openghg/store/base/_base.py
Comment (rt17603) [reply openghg/store/base/_base.py]: Should be done

File: openghg/util/_data_level.py
Comment (rt17603) [reply openghg/util/_data_level.py]: Should be done

File: openghg/store/base/_base.py
Comment (brendan-m-murphy) [reply openghg/store/base/_base.py]: Since python 3.9 "Dict" etc has been deprecated: https://peps.python.org/pep-0585/

I guess it doesn't really matter because at some point we'll probably want to use a code formatter to update all the type hints (e.g. https://pypi.org/project/pyupgrade/).

File: openghg/util/_data_level.py
Comment (brendan-m-murphy) [inline openghg/util/_data_level.py:12]: You could add int and float to the accepted inputs, the output will be str in any case. (With corresponding change in the actual implementation.) Not a big deal, I will probably look for things like that in the metadata formatting updates.

**Review by brendan-m-murphy — APPROVED** [review]
Looks good! I still have suggestions (opinions?) but they're just about type hints so I don't think it's worth holding up the PR. (PS let me know if you need a re-approval if you commit one of the dict vs Dict suggestions.)

File: openghg/store/base/_base.py
Comment (rt17603) [reply openghg/store/base/_base.py]: Ok - agreed this should be updated then in the longer term. For now I'll keep it as is to be consistent with the other uses of `data` but yes this should all get updated in one go really to replace `Dict` with `dict` where this is used.

**Pull request merged**
